packages/app/src/firebaseServerApp.ts

```diff
@@ -25,6 +25,77 @@ import { deleteApp } from './api';
 import { ComponentContainer } from '@firebase/component';
 import { FirebaseAppImpl } from './firebaseApp';
 import { ERROR_FACTORY, AppError } from './errors';
+import type { KeyObject } from 'node:crypto';
+
+let memoizedPublicKeys: Map<string, CryptoKey | KeyObject> | undefined;
+let memoizedPublicKeysExpireAt: Date | undefined;
+
+async function getPublicKeys(): Promise<Map<string, CryptoKey | KeyObject>> {
+  if (
+    memoizedPublicKeys &&
+    memoizedPublicKeysExpireAt &&
+    memoizedPublicKeysExpireAt < new Date()
```

Comment:
Shouldn't this be `memoizedPublicKeysExpireAt > new Date()` ?

Reply:
i think i'm going to move to a setTimeout anyway, would simply this logic a lot